src/hm/hmRadio.h

```diff
@@ -250,14 +265,22 @@ class HmRadio {
                 }
                 cnt++;
             }
-            sendPacket(invId, cnt, isRetransmit, isNoMI);
+            sendPacket(invId, txChan, cnt, isRetransmit, isNoMI);
         }
 
-        void prepareDevInformCmd(uint64_t invId, uint8_t cmd, uint32_t ts, uint16_t alarmMesId, bool isRetransmit, uint8_t reqfld=TX_REQ_INFO) { // might not be necessary to add additional arg.
+        void prepareDevInformCmd(uint64_t invId, inv_type_t invType, uint8_t txChan, uint8_t cmd, uint32_t ts, uint16_t alarmMesId, bool isRetransmit, uint8_t reqfld=TX_REQ_INFO) { // might not be necessary to add additional arg.
             if(mSerialDebug) {
                 DPRINT(DBG_DEBUG, F("prepareDevInformCmd 0x"));
                 DPRINTLN(DBG_DEBUG,String(cmd, HEX));
             }
+            uint8_t rxFrameCnt = 10;
+            if (cmd == RealTimeRunData_Debug) {
+                rxFrameCnt = INV_TYPE_HMCH1+1;
```

Comment:
Die max. Anzahl der erwarteten Antwortframes für INV_TYPE_HMCH2 ist 3 und für die INV_TYPE_HMCH4 ist die max. zu erwartende Anzahl 4. In diesem Code immer 2. Man sollte m.E. auch nicht mit der davon grundsätzlich unabhängigen Definition des enum invType rechnen, sondern das explizit ausformulieren.